### PR TITLE
Node consistency 51 60

### PR DESCRIPTION
**Pull request by martinstacey**
### Purpose
The following PR addresses nodes' names and descriptions to make them more descriptive for beginners.

**51) NODE: String.Replace**
-CHANGES: 
Change input and output from str to string

**52) NODE: String.Remove**
-CHANGES: 
Change input and output from str to string
-The count has no default value, so will eliminate description part where it specifies what happends if you dont have a value 

**53) NODE: String.Insert**
-CHANGES: 
Change input and output from str to string

**54) NODE: String.AllIndicesOf**
-CHANGES: 
Change input and output from str to string
Description: Finds a list of zero-based indexes of occurrence of a sub-string inside a string.
InputDescriptions:
string: A string to search in.
searchFor: Substring to search for.
ignoreCase: Whether or not comparison takes case into account.
Output Description:
indexList: List of indices or ints where substring is found
Search: all indices of,find substring,where,search

**55) NODE: String.Join**
-CHANGES: 
Change output from str to string

**56) NODE: String.Concat**
-CHANGES: 
Change output from str to string

**57) NODE: Python Script From String**
No changes needed

**58) NODE: Python Script**
No changes needed

**59) NODE: ScopeIf**
CHANGES:
Description: Returns the input of either the True or False input depending on what boolean value is toggled in the test input.
Input Description: 
Test: Boolean test
True: Returned if test is true
False: Returned if test is false

**60) NODE: LoopWhile**
CHANGES:
Inputs:
name initialValue
description: initial value for loop</param>
Outpus:
name: result
description: result from loop

**61) NODE: If**
CHANGES:
Description: Returns the input of either the True or False input depending on what boolean value is toggled in the test input.
Input Description: 
Test: Boolean test
True: Returned if test is true
False: Returned if test is false

image

### Reviewers
@SHKnudsen
@StudioLE
@MarkThorley
@anrova

### FYIs
-No FYI

### CODE REVIEW

**Review by QilongTang — APPROVED** [review]

File: src/DynamoCore/BuiltInAndOperators/BuiltIn.xml [hunk 1]
```diff
@@ -157,9 +157,10 @@
     </member>
     <member name="LoopWhile">
       <summary>Constructs a while loop node given an initial value, a condition and a loop body.</summary>
-      <param name="init">initial value</param>
+      <param name="initialValue">initial value for loop</param>
       <param name="continueWhile">while loop condition</param>
       <param name="loopBody">while loop body</param>
+      <returns name="result">result from loop</returns>
```
Comment (aparajit-pratap) [inline src/DynamoCore/BuiltInAndOperators/BuiltIn.xml:163]: `var[]..[]` type in description?

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx [hunk 2]
```diff
@@ -447,7 +447,7 @@
     <value>Starting accumulated value, to be passed into the first call to the Reductor function.</value>
   </data>
   <data name="ScopeIfDescription" xml:space="preserve">
-    <value>Scoped If statement</value>
+    <value>Returns the input of either the True or False input depending on what boolean value is toggled in the test input.</value>
```
Comment (aparajit-pratap) [inline src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx:450]: `Returns the result of either the ...`

File: src/Libraries/CoreNodeModels/Properties/Resources.resx [hunk 3]
```diff
@@ -447,7 +447,7 @@
     <value>Starting accumulated value, to be passed into the first call to the Reductor function.</value>
   </data>
   <data name="ScopeIfDescription" xml:space="preserve">
-    <value>Scoped If statement</value>
+    <value>Returns the input of either the True or False input depending on what boolean value is toggled in the test input.</value>
```
Comment (aparajit-pratap) [inline src/Libraries/CoreNodeModels/Properties/Resources.resx:450]: `Returns the result of either ... depending on the boolean value of the test input`

File: src/Libraries/CoreNodes/String.cs [hunk 4]
```diff
@@ -310,15 +310,22 @@ public static int IndexOf(string str, string searchFor, bool ignoreCase = false)
...
         }
-
-        public static int[] AllIndicesOf(string str, string searchFor, bool ignoreCase = false)
+        /// <summary>
+        /// Finds list of indices where sub-string appears inside a string.
+        /// </summary>
+        /// <param name="string">A string to search in.</param>
+        /// <param name="searchFor">Substring to search for.</param>
+        /// <param name="ignoreCase">Whether or not comparison takes case into account.</param>
+        /// <returns name="indexList">List of indices (integers) where substring is found</returns>
```
Comment (aparajit-pratap) [inline src/Libraries/CoreNodes/String.cs:319]: Does this hide the return type information?

File: src/DynamoCore/BuiltInAndOperators/BuiltIn.xml (hunk 1 above)
Comment (martinstacey) [reply src/DynamoCore/BuiltInAndOperators/BuiltIn.xml:163]: image

@aparajit-pratap  do you know if LoopWhile works for lists? I wasn't able to make it work, if it doesn't we should rename the output description to: 
`type: number (result from loop)`

File: src/Libraries/CoreNodeModels/Properties/Resources.resx (hunk 3 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.resx]: addressed in bd2d1cb

File: src/Libraries/CoreNodes/String.cs (hunk 4 above)
Comment (martinstacey) [reply src/Libraries/CoreNodes/String.cs:319]: @aparajit-pratap I changed the output name to be a little bit more descriptive, do you think I should revert the name to `int[]` ?
image

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx (hunk 2 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx]: Addressed in bd2d1cb

File: src/DynamoCore/BuiltInAndOperators/BuiltIn.xml (hunk 1 above)
Comment (martinstacey) [reply src/DynamoCore/BuiltInAndOperators/BuiltIn.xml]: Addressed in 4dc47fd

File: src/Libraries/CoreNodes/String.cs (hunk 4 above)
Comment (Amoursol) [reply src/Libraries/CoreNodes/String.cs:319]: @martinstacey Please do retain descriptive output ports. @aparajit-pratap, correct me if I'm wrong, but you just also wanted to add the `in[]` to it?

**Review by Amoursol — APPROVED** [review]

File: src/Libraries/CoreNodes/String.cs (hunk 4 above)
Comment (aparajit-pratap) [reply src/Libraries/CoreNodes/String.cs:319]: @Amoursol that's correct, apologize for not being clearer.

Comment (aparajit-pratap) [reply src/Libraries/CoreNodes/String.cs:319]: @martinstacey please add `int[]` to the description:
`type: int[], List of indices where substring is found`

Comment (martinstacey) [reply src/Libraries/CoreNodes/String.cs]: Addressed in 926c7cb

**Pull request merged**